lib/DXIL/DxilShaderFlags.cpp

```diff
@@ -373,6 +373,7 @@ ShaderFlags ShaderFlags::CollectShaderFlags(const Function *F,
   M->GetValidatorVersion(valMajor, valMinor);
   bool hasMulticomponentUAVLoadsBackCompat = valMajor == 1 && valMinor == 0;
   bool hasViewportOrRTArrayIndexBackCombat = valMajor == 1 && valMinor < 4;
+  bool hasBarycentricsBackCompat = valMajor == 1 && valMinor < 6;
```

Comment:
**Unrelated**, but this reminded me of a use case I've recently become aware of where compilation is performed in one step and validation and signing in another. In such a case, it is possible that the internal validator version at the compile step differs from that of the post-compile validation version. Naturally, this is pre-existing, so this is not a new problem if it is.

We have the `-dxil-version` flag which should probably be used in this use case. Maybe we should make it more visible.

Reply:
That's what -validator-version is for.  It's not an option we would expect people to use normally unless they really know what they are doing though.

Reply:
`-validator-version` is what I meant to say. No real action required here. just ruminating